### PR TITLE
pynitrokey: fix build and update to 0.4.31

### PR DESCRIPTION
**Pull request by jojosch**
###### Description of changes

Got my Nitrokey 3 today but wasn't able to build `pynitrokey`:

```
ERROR: Could not find a version that satisfies the requirement protobuf<4.0.0,>=3.17.3 (from nrfutil) (from versions: none)
ERROR: No matching distribution found for protobuf<4.0.0,>=3.17.3
```

Updated `pynitrokey` to the latest release and updated its dependencies.

-------------------

Changelog for `pynitrokey`:
https://github.com/Nitrokey/pynitrokey/releases/tag/v0.4.28.nitrokey ([changes](https://github.com/Nitrokey/pynitrokey/compare/v0.4.27.nitrokey...v0.4.28.nitrokey))
https://github.com/Nitrokey/pynitrokey/releases/tag/v0.4.29.nitrokey ([changes](https://github.com/Nitrokey/pynitrokey/compare/v0.4.28.nitrokey...v0.4.29.nitrokey))
https://github.com/Nitrokey/pynitrokey/releases/tag/v0.4.30.nitrokey ([changes](https://github.com/Nitrokey/pynitrokey/compare/v0.4.29.nitrokey...v0.4.30.nitrokey))
https://github.com/Nitrokey/pynitrokey/releases/tag/v0.4.31.nitrokey ([changes](https://github.com/Nitrokey/pynitrokey/compare/v0.4.30.nitrokey...v0.4.31.nitrokey))

new dependency with 0.4.31: https://github.com/jlusiardi/tlv8_python

-------------------

Now pynitropy works:

```
$ ./result/bin/nitropy list
Command line tool to interact with Nitrokey devices 0.4.31
:: 'Nitrokey FIDO2' keys
:: 'Nitrokey Start' keys:
:: 'Nitrokey 3' keys
/dev/hidraw1: Nitrokey 3 ....

$ ./result/bin/nitropy nk3 rng
Command line tool to interact with Nitrokey devices 0.4.31
32d098b4d27ab74f217864d67a583bed31560f28b3bf755a5....

./result/bin/nitropy nk3 test
Command line tool to interact with Nitrokey devices 0.4.31
Found 1 Nitrokey 3 device(s):
- Nitrokey 3 at /dev/hidraw1

Running tests for Nitrokey 3 at /dev/hidraw1

[1/3]	uuid     	UUID query              	SUCCESS  	....
[2/3]	version  	Firmware version query  	SUCCESS  	v1.2.2
Please press the touch button on the device ...
Please press the touch button on the device ...
[3/3]	fido2    	FIDO2                   	SUCCESS  	

3 tests, 3 successful, 0 skipped, 0 failed

Summary: 1 device(s) tested, 1 successful, 0 failed
```

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/click-command-tree/default.nix [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  ];
+
+  pytestFlagsArray = [ "tests.py" ];
+
+  pythonImportsCheck = [
+    "click_command_tree"
+  ];
+
+  meta = with lib; {
+    description = "click plugin to show the command tree of your CLI";
```
Comment (mattmelling) [inline pkgs/development/python-modules/click-command-tree/default.nix:46]: c -> C

File: pkgs/development/python-modules/tlv8/default.nix [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  checkInputs = [
+    pytestCheckHook
+  ];
+
+  pythonImportsCheck = [
+    "tlv8"
+  ];
+
+  meta = with lib; {
+    description = "Type-Length-Value8 (TLV8) for python";
```
Comment (mattmelling) [inline pkgs/development/python-modules/tlv8/default.nix:29]: python -> Python

**Review by mattmelling — CHANGES_REQUESTED** [review]
Looks OK so far, however unable to test as there are merge conflicts, please rebase on to a newer master.

File: pkgs/development/python-modules/click-command-tree/default.nix (hunk 1 above)
Comment (jojosch) [reply pkgs/development/python-modules/click-command-tree/default.nix:46]: This was added to nixpkgs in the meantime in #204536. Should I fix just the typo in this PR or should we wait for the next  update of `click-command-tree` and fix the typo in that PR?

**Comment by jojosch** [comment]
Rebased on master. The `spsdk` update and addition of `click-command-tree` was done in #204536.

File: pkgs/development/python-modules/click-command-tree/default.nix (hunk 1 above)
Comment (mattmelling) [reply pkgs/development/python-modules/click-command-tree/default.nix]: Not a big deal if it's already been approved and merged.

**Comment by mattmelling** [comment]
Result of `nixpkgs-review pr 195816` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>pynitrokey</li>
    <li>python310Packages.tlv8</li>
    <li>python39Packages.tlv8</li>
  </ul>
</details>

**Review by mattmelling — APPROVED** [review]

**Review by frogamic — APPROVED** [review]

**Pull request merged**